src/main/java/de/unistuttgart/iste/gits/skilllevel_service/service/calculation/SkillLevelCalculator.java

```diff
@@ -34,34 +32,32 @@ public class SkillLevelCalculator {
      */
     public AllSkillLevelsEntity recalculateLevels(UUID chapterId, UUID userId,
                                                   AllSkillLevelsEntity allSkillLevelsEntity) {
-        List<Content> contents = contentServiceClient.getContentsWithUserProgressData(userId, List.of(chapterId));
+        List<ContentServiceClient.GenericAssessmentResponse> assessments =
+                contentServiceClient.getAssessmentsOfChapter(userId, chapterId);
 
         // set all skill levels to 0 and clear their logs. We're recalculating everything from scratch
         allSkillLevelsEntity.setAnalyze(new SkillLevelEntity(0));
         allSkillLevelsEntity.setRemember(new SkillLevelEntity(0));
         allSkillLevelsEntity.setUnderstand(new SkillLevelEntity(0));
         allSkillLevelsEntity.setApply(new SkillLevelEntity(0));
 
-        return calculate(allSkillLevelsEntity, contents);
+        return calculate(allSkillLevelsEntity, assessments);
     }
 
-    private AllSkillLevelsEntity calculate(AllSkillLevelsEntity allSkillLevelsEntity, List<Content> contents) {
-        if(contents.isEmpty()) {
+    private AllSkillLevelsEntity calculate(AllSkillLevelsEntity allSkillLevelsEntity,
+                                           List<ContentServiceClient.GenericAssessmentResponse> assessments) {
```

Comment:
I don't really like that you use the rather internal mapping class here outside of the client class. I personally would make "GenericAssessmentResponse" implement the "Assessment" interface, then you can just use Assessment here

Reply:
yes